libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.h

```diff
@@ -61,6 +61,13 @@
  */
 + (void) removeSharedInstanceWithUser:(nonnull SFUserAccount *) userAccount;
 
+/**
+ * Returns an instance of this class associated with an unauthenticated context (no authenticated user account).
+ *
+ * @return Instance of this class.
+ */
++ (nonnull instancetype) sharedUnauthenticatedInstance;
```

Comment:
Could we add a `removeSharedUnauthenticatedInstance` convenience method as well?

Reply:
I can do that.  I wasn't sure if there was value, given that it's just a toggle in memory—there are no constraints on creating / utilizing the object.

Reply:
@bhariharan We talked offline about the value of removal.  Looking through the code for authenticated analytics managers, removing the manager singleton doesn't do anything other than remove it from memory.  The managers rely upon the authentication notifications to remove events from their stores.

Seems like that's probably a better way to go: explicit store clearing, vs. object manipulation.  What do you think?

Reply:
(These are ultimately fronted by the `[SFSDKAnalyticsManager reset]` method.)